api/server/call_logs.go

```diff
@@ -21,18 +29,37 @@ func (s *Server) handleCallLogGet(c *gin.Context) {
 		return
 	}
 
-	// TODO this API needs to change to text/plain / gzip anyway, punting
-	// optimization, but we can write this direct to the wire, too... seems like
-	// we should write some kind of writev json thing for go since we keep
-	// hitting this :(
-	var b bytes.Buffer
-	b.ReadFrom(logReader)
-
-	callObj := models.CallLog{
-		CallID:  callID,
-		AppName: appName,
-		Log:     b.String(),
-	}
+	mimeTypes, _ := c.Request.Header["Accept"]
 
-	c.JSON(http.StatusOK, callLogResponse{"Successfully loaded log", &callObj})
+	for _, mimeType := range mimeTypes {
+		if strings.Contains(mimeType, "application/json") {
+			var b bytes.Buffer
+			b.ReadFrom(logReader)
+			c.JSON(http.StatusOK, callLogResponse{"Successfully loaded log",
+				&models.CallLog{
+					CallID:  callID,
+					AppName: appName,
+					Log:     b.String(),
+				}})
+			return
+		}
+		if strings.Contains(mimeType, "text/plain") {
+			io.Copy(c.Writer, logReader)
+			return
+
+		}
+		if strings.Contains(mimeType, "*/*") {
+			var b bytes.Buffer
+			b.ReadFrom(logReader)
+			c.JSON(http.StatusOK, callLogResponse{"Successfully loaded log",
+				&models.CallLog{
+					CallID:  callID,
+					AppName: appName,
+					Log:     b.String(),
+				}})
+			return
+		}
+	}
+	// if we've reached this point it means that Fn didn't recognize Accepted content type
+	c.Status(http.StatusNotAcceptable)
```

Comment:
if no accept headers are given, we also want to return a json. this is only if they provide accept headers, none of which we can offer.

p.s. extract the function for the json logic, please

Reply:
please add a body to this response, as well